File: cs214/projects/project06/array.rb
# array.rb takes a specified number of inputs from the  
# user and prints them back out.
#
# Begun by:Charles Blum, for CS 214 at Calvin College.
# Completed by:: Charles Blum
# Date: 02 April 2014
########################################################

###############################################
# readArray reads in the values from the user
# Receive: the array of size 0.
################################################
def readArray( someArray, size ) 
  for i in 0..(size-1)
    someArray.push( gets.to_f )
  end
  someArray
end 

###############################################
# wrietArray writes out the array to the user
# Receive: the array of size n.
################################################
def writeArray(someArray)
  someArray.each { |i| puts( i ) }
end

def main
  puts "enter the size of the array: "
  size = gets.to_i

  puts "enter #{size} values: "
  someArray = []

  readArray( someArray, size )
  writeArray( someArray )
end

main

File: cs214/projects/project06/array.cpp
/* array.cpp creates an array of size arraySize
 * taken from the user and then fills it with 
 * values from the user.
 *
 * Begun by: Charles Blum, for CS 214 at Calvin College.
 * Completed by:Charles Blum
 * Date: 02 April 2014
 */

#include <iostream>
#include <numeric>
using namespace std;

void printArray(double anArray[], int itsSize, ostream & out);
void readArray(double anArray[], int itsSize, istream & in);

int main() {
 int size;

  cout << "\n specify an array size: " << flush;
  cin >> size;
 
  // Define array0 as an array containing no values
  double array[size];
 
  readArray(array, size, cin);
  printArray(array, size, cout);
}

/*******************************************************
 * readArray takes in itsSize inputs and places them   *
 * in an array.                                        *
 * Receive: anArray, an empty array of doubles.        *
 * 	    itsSize, the size of the array             *
 *          istream, where to read from.               *
 *******************************************************/
void readArray(double anArray[], int itsSize, istream & in = cin) {
  for ( int i = 0; i < itsSize; i++ )
  {
  in >> anArray[i];
  }
}

/*******************************************************
 * printArray prints the values in the arrya           *
 * Receive: anArray, an array of doubles.              *
 * 	    itsSize, the size of the array             *
 *          ostream, where to write to.                *
 *******************************************************/
void printArray(double anArray[], int itsSize, ostream & out = cout) {
  for ( int i = 0; i < itsSize; i++ )
  {
  out << anArray[i] << " " << flush;
  }
  out << endl;
}
